tests/bsim/bluetooth/samples/battery_service/src/central_test.c

```diff
@@ -402,6 +447,30 @@ static void subscribe_battery_level(const struct bt_gatt_attr *attr)
 	read_battery_level(attr);
 }
 
+static void subscribe_battery_critical_status(const struct bt_gatt_attr *attr)
+{
+	int err;
+
+	battery_critical_status_sub_params = (struct bt_gatt_subscribe_params){
+		/* In Zephyr, it is common practice for the CCC handle
+		 * to be positioned two handles after the characteristic handle.
+		 */
+		.ccc_handle = bt_gatt_attr_get_handle(attr) + 2,
+		.value_handle = bt_gatt_attr_value_handle(attr),
+		.value = BT_GATT_CCC_INDICATE,
+		.notify = battery_critical_status_indicate_cb,
+	};
+
+	err = bt_gatt_subscribe(default_conn, &battery_critical_status_sub_params);
+	if (err && err != -EALREADY) {
+		TEST_FAIL("Subscribe failed (err %d)\n", err);
+	} else {
+		LOG_DBG("Battery critical status [SUBSCRIBED]\n");
+	}
+
+	bt_gatt_read(default_conn, &read_bcs_params);
```

Comment:
Why are we performing a read in this function *and* in `test_bas_central_main`? 

Suggest to follow the same pattern as you implemented for battery level (and/or modify the battery level to follow the same pattern as BCS)

Reply:
Used the same pattern as BCS